tests/lib/rules/no-constant-condition.js

```diff
@@ -58,7 +58,15 @@ ruleTester.run("no-constant-condition", rule, {
         // { checkLoops: false }
         { code: "while(true);", options: [{ checkLoops: false }] },
         { code: "for(;true;);", options: [{ checkLoops: false }] },
-        { code: "do{}while(true)", options: [{ checkLoops: false }] }
+        { code: "do{}while(true)", options: [{ checkLoops: false }] },
+
+        { code: "function* foo(){while(true){yield 'foo';}}", parserOptions: { ecmaVersion: 6 } },
+        { code: "function* foo(){for(;true;){yield 'foo';}}", parserOptions: { ecmaVersion: 6 } },
+        { code: "function* foo(){do{yield 'foo';}while(true)}", parserOptions: { ecmaVersion: 6 } },
+        { code: "function* foo(){while (true) { while(true) {yield;}}}", parserOptions: { ecmaVersion: 6 } },
+        { code: "function* foo() {for (; yield; ) {}}", parserOptions: { ecmaVersion: 6 } },
+        { code: "function* foo() {for (; ; yield) {}}", parserOptions: { ecmaVersion: 6 } },
+        { code: "function* foo() {while (true) {function* foo() {yield;}yield;}}", parserOptions: { ecmaVersion: 6 } }
```

Comment:
Nitpick: Instead of specifying `parserOptions: { ecmaVersion: 6 }` for each of these tests, it would be better to add it to the `RuleTester` constructor:

```js
const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 6 } });
```